File: 2868-continuous-subarrays/2868-continuous-subarrays.java
import java.util.Deque;
import java.util.LinkedList;

class Solution {
    public long continuousSubarrays(int[] nums) {
        int n = nums.length;
        int st = 0;
        long count = 0;

        Deque<Integer> i = new LinkedList<>();
        Deque<Integer> d = new LinkedList<>();

        for (int j = 0; j < n; j++) {
            while (!i.isEmpty() && nums[j] < nums[i.peekLast()]) {
                i.pollLast();
            }
            while (!d.isEmpty() && nums[j] > nums[d.peekLast()]) {
                d.pollLast();
            }

            i.addLast(j);
            d.addLast(j);

            while (nums[d.peekFirst()] - nums[i.peekFirst()] > 2) {
                if (i.peekFirst() == st) {
                    i.pollFirst();
                }
                if (d.peekFirst() == st) {
                    d.pollFirst();
                }
                st++;
            }

            count += (j - st + 1);
        }

        return count;
    }
}

File: 2868-continuous-subarrays/2868-continuous-subarrays.py
from collections import deque

class Solution:
    def continuousSubarrays(self, nums: List[int]) -> int:
        n = len(nums)
        left = 0
        count = 0
        min_q = deque()
        max_q = deque()

        for right in range(n):
            while min_q and nums[right] < min_q[-1]:
                min_q.pop()
            while max_q and nums[right] > max_q[-1]:
                max_q.pop()
            
            min_q.append(nums[right])
            max_q.append(nums[right])

            while max_q[0] - min_q[0] > 2:
                if nums[left] == min_q[0]:
                    min_q.popleft()
                if nums[left] == max_q[0]:
                    max_q.popleft()
                left += 1
            
            count += right - left + 1
        
        return count
